openmc/data/photonuclear.py

```diff
@@ -0,0 +1,879 @@
+        if hasattr(self, '_evaluation'):
+            raise NotImplementedError('Cannot export incident photonuclear data that '
+                                      'originated from an ENDF file.')
+        # Write basic data
+        g = f.create_group(self.name)
+        g.attrs['Z'] = self.atomic_number
+        g.attrs['A'] = self.mass_number
+        g.attrs['metastable'] = self.metastable
+        g.attrs['atomic_weight_ratio'] = self.atomic_weight_ratio
+
+        # Determine union energy grid
+        union_grid = np.array([])
+        for rx in self:
+            union_grid = np.union1d(union_grid, rx.xs.x)
+            for product in rx.products:
+                union_grid = np.union1d(union_grid, product.yield_.x)
+        g.create_dataset("energy", data=union_grid)
+        
+        # Write reaction data
+        rxs_group = g.create_group('reactions')
+        for rx in self.reactions.values():
+            # Skip writing redundant reaction if it doesn't have neutron
+            # production or is a summed transmutation reaction.
+            # Also write heating.
+            if rx.redundant:
+                neutron_rx = any(p.particle == 'neutron' for p in rx.products)
+                keep_mts = (301,)
+                if not (neutron_rx or rx.mt in keep_mts):
+                    continue
+
+            rx_group = rxs_group.create_group(f'reaction_{rx.mt:03}')
+            rx.to_hdf5(rx_group)
+        
+        # Write fission energy release data
+        if self.fission_energy is not None:
+            fer_group = g.create_group('fission_energy_release')
+            self.fission_energy.to_hdf5(fer_group)
+        
+        f.close()
+
+    @classmethod
+    def from_hdf5(cls, group_or_filename):
+        """Generate photo-nuclear interaction data from HDF5 group
+
+        Parameters
+        ----------
+        group_or_filename : h5py.Group or str
+            HDF5 group containing interaction data. If given as a string, it is
+            assumed to be the filename for the HDF5 file, and the first group is
+            used to read from.
+
+        Returns
+        -------
+        openmc.data.Photonuclear
+            photo-nuclear interaction data
+
+        """
+        if isinstance(group_or_filename, h5py.Group):
+            group = group_or_filename
+        else:
+            h5file = h5py.File(str(group_or_filename), "r")
+
+            # Make sure version matches
+            if "version" in h5file.attrs:
+                major, minor = h5file.attrs["version"]
+                # For now all versions of HDF5 data can be read
+            else:
+                raise IOError(
+                    "HDF5 data does not indicate a version. Your installation of "
+                    "the OpenMC Python API expects version {}.x data.".format(
+                        HDF5_VERSION_MAJOR
+                    )
+                )
+
+            group = list(h5file.values())[0]
```

Comment:
Leaving file handles open can lead to Too many open files errors and other resource-related problems, especially when processing many files in a script. 
```suggestion
            group = list(h5file.values())[0]
            h5file.close()
```